lib/actions/getVariantsBySku.js

```diff
@@ -0,0 +1,99 @@
+"use strict";
+const request        = require('request');
+const requestOptions = require('../requestOptions.js');
+const handleResponse = require('../handleResponse.js');
+const messages = require('elasticio-node').messages;
+
+exports.process = processAction;
+
+function processAction(message, config) {
+  let self = this;
+
+  // submits the HTTP request to the VMS to fulfil the order
+  function getVariantsBySku(authenticationHeaders) {
+    let skus = message.body.currentMessage;
+
+    skus = skus.map(sku => sku.sku).join(',');
+
+    let options = Object.assign({}, authenticationHeaders, {
+      json: true,
+      url: `${authenticationHeaders.url}variants?filter[sku]=${skus}&fields[variants]=sku,inventory-quantity`
+    });
+
+    return new Promise((resolve, reject) => {
+      request.get(options, (error, response, body) => {
+        if (error) {
+          reject(error);
+        } else {
+          resolve(response);
+        }
+      });
+    });
+  }
+
+  function filterVariants(response) {
+    let skus = message.body.currentMessage;
+    let variants = response.body.data;
+
+    let filteredVariants = variants.reduce((variants, variant) => {
```

Comment:
I might be wrong here but I think this could be just done with `filter` instead.
```
let filteredVariants = variants.filter(variant => {
  let sku = skus.find(sku => sku.sku == variant.attributes.sku);
  return (sku && variant.attributes['inventory-quantity'] != sku.quantity);
});